### ISSUE TITLE
训练时，label与logits第一维度不匹配

### ISSUE THREAD

**Issue by ComingLeaves** [yanwii/dynamic-seq2seq#25]
在进行第一轮训练时，当step101结束后报错：logits and labels must have the same first dimension, got logits shape [2800,9003] and labels shape [2860]，不知道该怎么调整代码的label与logits长度相等。

**Comment by dengxiaotian123** [comment]
请问怎么解决的，我和你遇到一样的问题

**Comment by ComingLeaves** [comment]
> 请问怎么解决的，我和你遇到一样的问题
至今依旧没有解决，我尝试更改代码，并按照网络论坛给出的方法，截断logits与label长度一致，但很遗憾没有起到作用。

**Comment by yanwii** [comment]
试着调整 **encoder_vocab_size**, **decoder_vocab_size**这两个参数

**Comment by dengxiaotian123** [comment]
> 试着调整 **encoder_vocab_size**, **decoder_vocab_size**这两个参数

self.encoder_vocab_size = data_map.get("Q_vocab_size")是根据词表的个数来的，并非自行设置的，应该是没有错的吧，只要训练step超过100就会报错。训练到101 就会报错

**Comment by ComingLeaves** [comment]
> > 试着调整 **encoder_vocab_size**, **decoder_vocab_size**这两个参数
> 
> self.encoder_vocab_size = data_map.get("Q_vocab_size")是根据词表的个数来的，并非自行设置的，应该是没有错的吧，只要训练step超过100就会报错。训练到101 就会报错

动态lstm里有encoder_vocab_size和decoder_vocab_size两参数，但是不知道具体意义，seq2seq文件的初始化里的两个参数是根据预料分词来的，不能做改变。我看到后面进行序列编码时，用的是它的长度，最后得出来logits与label长度不一致报错。昨天调过两个参数，依旧还是老样子。

**Comment by yanwii** [comment]
请问一下你们的TensorFlow是什么版本的。

**Comment by ComingLeaves** [comment]
> 请问一下你们的TensorFlow是什么版本的。

Tensorflow-GPU 1.5

**Comment by ComingLeaves** [comment]
请问，你在嵌入层是如何实现中文序列化？独热编码还是其他的方式，我看到生成的文本内容不像是one hot矩阵，像是16进制

**Comment by zx1097133696** [comment]
> > 试着调整**encoder_vocab_size**，**decoder_vocab_size**这两个参数
> 
> self.encoder_vocab_size = data_map.get（“ Q_vocab_size”）是根据词表的个数来的，而不是永久设置的，应该是没有错的吧，只要训练step超过100就会报错。训练到101就会报错

解决这个问题了嘛

**Comment by ComingLeaves** [comment]
没有




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "yanwii/dynamic-seq2seq"                                                                                    <notifications@github.com&gt;;
发送时间:&nbsp;2020年11月27日(星期五) 下午2:57
收件人:&nbsp;"yanwii/dynamic-seq2seq"<dynamic-seq2seq@noreply.github.com&gt;;
抄送:&nbsp;"纹"<1554091920@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [yanwii/dynamic-seq2seq] 训练时，label与logits第一维度不匹配 (#25)





   
试着调整encoder_vocab_size，decoder_vocab_size这两个参数
  
self.encoder_vocab_size = data_map.get（“ Q_vocab_size”）是根据词表的个数来的，而不是永久设置的，应该是没有错的吧，只要训练step超过100就会报错。训练到101就会报错
  
解决这个问题了嘛
